### PR TITLE
chore: support go-junit-report

### PR DESCRIPTION
**Pull request by ghaiszaher**
### What
* Support test JUnit reports generated by https://github.com/jstemmer/go-junit-report
* Tested in https://github.com/ScaCap/pdf-service/pull/726

### How
* Pass new argument in case filename is mentioned in stack trace instead of `class` attribute
* Add tests

### CODE REVIEW

File: utils.js [hunk 1]
```diff
@@ -4,23 +4,30 @@ const fs = require('fs');
...
     const matches = output.match(new RegExp(`${filename}.*?:\\d+`, 'g'));
-    if (!matches) return { filename: filename, line: 1 };
+    console.log({matches});
+    if (!matches) return {filename: filename, line: 1};
 
     const [lastItem] = matches.slice(-1);
     const [, line] = lastItem.split(':');
-    core.debug(`Resolved file ${filename} and line ${line}`);
-
+    core.info(`Resolved file ${filename} and line ${line}`);
```
Comment (ghaiszaher) [inline utils.js:22]:
```suggestion
    core.debug(`Resolved file ${filename} and line ${line}`);

```

File: utils.js [hunk 2]
```diff
@@ -4,23 +4,30 @@ const fs = require('fs');
...
+    if (isFilenameInOutput) {
+        filename = output.split(':')[0].trim();
+    } else {
+        filename = file ? file : classname.split('.').slice(-1)[0].split('(')[0];
+    }
+    console.log({file, filename, classname, output});
     const matches = output.match(new RegExp(`${filename}.*?:\\d+`, 'g'));
-    if (!matches) return { filename: filename, line: 1 };
+    console.log({matches});
+    if (!matches) return {filename: filename, line: 1};
```
Comment (ghaiszaher) [inline utils.js:18]:
```suggestion
    if (!matches) return { filename: filename, line: 1 };
```

File: utils.js [hunk 3]
```diff
@@ -4,23 +4,30 @@ const fs = require('fs');
...
-    core.debug(`Resolved file ${filename} and line ${line}`);
-
+    core.info(`Resolved file ${filename} and line ${line}`);
     return { filename, line: parseInt(line) };
 };
 
 const resolvePath = async filename => {
-    core.debug(`Resolving path for ${filename}`);
-    const globber = await glob.create(`**/${filename}.*`, { followSymbolicLinks: false });
+    core.info(`Resolving path for ${filename}`);
```
Comment (ghaiszaher) [inline utils.js:27]:
```suggestion
    core.debug(`Resolving path for ${filename}`);
```

File: utils.js [hunk 4]
```diff
@@ -4,23 +4,30 @@ const fs = require('fs');
...
+    const globber = await glob.create([`**/${filename}.*`, `**/${filename}`].join('\n'), { followSymbolicLinks: false });
     const results = await globber.glob();
-    core.debug(`Matched files: ${results}`);
+    core.info(`Matched files: ${results}`);
```
Comment (ghaiszaher) [inline utils.js:30]:
```suggestion
    core.debug(`Matched files: ${results}`);
```

File: utils.js [hunk 5]
```diff
@@ -32,17 +39,17 @@ const resolvePath = async filename => {
     } else {
         path = filename;
     }
-    core.debug(`Resolved path: ${path}`);
+    core.info(`Resolved path: ${path}`);
```
Comment (ghaiszaher) [inline utils.js:42]:
```suggestion
    core.debug(`Resolved path: ${path}`);
```

File: utils.js [hunk 6]
```diff
@@ -32,17 +39,17 @@ const resolvePath = async filename => {
...
 
     // canonicalize to make windows paths use forward slashes
     const canonicalPath = path.replace(/\\/g, '/');
-    core.debug(`Canonical path: ${canonicalPath}`);
+    core.info(`Canonical path: ${canonicalPath}`);
```
Comment (ghaiszaher) [inline utils.js:46]:
```suggestion
    core.debug(`Canonical path: ${canonicalPath}`);
```

File: utils.js [hunk 7]
```diff
@@ -32,17 +39,17 @@ const resolvePath = async filename => {
...
 
     return canonicalPath;
 };
 
 async function parseFile(file) {
-    core.debug(`Parsing file ${file}`);
+    core.info(`Parsing file ${file}`);
```
Comment (ghaiszaher) [inline utils.js:52]:
```suggestion
    core.debug(`Parsing file ${file}`);
```

File: utils.js [hunk 8]
```diff
@@ -66,6 +73,8 @@ async function parseFile(file) {
             count++;
             if (testcase.skipped) skipped++;
             if (testcase.failure || testcase.flakyFailure || testcase.error) {
+                console.log('testcase:');
+                console.log(testcase);
```
Comment (ghaiszaher) [inline utils.js:77]:
```suggestion
```

File: utils.js [hunk 9]
```diff
@@ -76,7 +85,7 @@ async function parseFile(file) {
                     '';
                 testcaseData = Array.isArray(testcaseData) ? testcaseData : [testcaseData];
                 const stackTrace = (testcaseData.length ? testcaseData.join('') : '').trim();
-
+                core.info(`stackTrace: ${stackTrace}`)
```
Comment (ghaiszaher) [inline utils.js:88]:
```suggestion

```

File: utils.js [hunk 10]
```diff
@@ -89,6 +98,7 @@ async function parseFile(file) {
                         testcase.error._attributes.message) ||
                     stackTrace.split('\n').slice(0, 2).join('\n')
                 ).trim();
+                core.info(`message: ${message}`)
```
Comment (ghaiszaher) [inline utils.js:101]:
```suggestion
```

File: utils.js [hunk 11]
```diff
@@ -124,6 +134,7 @@ const parseTestReports = async reportPaths => {
     let skipped = 0;
     for await (const file of globber.globGenerator()) {
         const { count: c, skipped: s, annotations: a } = await parseFile(file);
+        core.info(JSON.stringify({count, skipped, annotations}));
```
Comment (ghaiszaher) [inline utils.js:137]:
```suggestion
```

File: utils.js [hunk 12]
```diff
@@ -4,8 +4,15 @@ const fs = require('fs');
 const parser = require('xml-js');
 
 const resolveFileAndLine = (file, classname, output) => {
+    const isFilenameInOutput = core.getInput('file_name_in_stack_trace') === 'true';
```
Comment (ghaiszaher) [inline utils.js:7]: Maybe this should be moved up to `action.js`

File: utils.js [hunk 13]
```diff
@@ -4,8 +4,15 @@ const fs = require('fs');
...
     // extract filename from classname and remove suffix
-    const filename = file ? file : classname.split('.').slice(-1)[0].split('(')[0];
+    let filename;
+    if (isFilenameInOutput) {
+        filename = output.split(':')[0].trim();
+    } else {
+        filename = file ? file : classname.split('.').slice(-1)[0].split('(')[0];
+    }
```
Comment (ghaiszaher) [inline utils.js:14]: Missing test

**Review by crypticmind — APPROVED** [review]

**Review by renatoabreu11 — APPROVED** [review]

**Review by sutil-dyego — APPROVED** [review]

**Pull request merged**